internal/servers/worker/internal/metric/proxy_ws.go

```diff
@@ -0,0 +1,118 @@
+package metric
+
+import (
+	"bufio"
+	"net"
+	"net/http"
+
+	"github.com/hashicorp/boundary/globals"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	proxyWebsocketSubsystem = "proxy_websocket"
+)
+
+var (
+	msgSizeBuckets = prometheus.ExponentialBuckets(100, 10, 8)
+
+	activeConnCount = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: globals.MetricNamespace,
+			Subsystem: proxyWebsocketSubsystem,
+			Name:      "active_connection_count",
+			Help:      "Count of open websocket connections to Boundary workers.",
+		},
+	)
+
+	bytesReceived = prometheus.NewHistogram(
```

Comment:
FYI, the bytes sent/received should be a counter, not a histogram.  You can see the relevant part of the RFC, but in summary, histograms make sense if there is a natural observation time to record the observation for.  Since this is a continual stream, that doesn't exist here.  Instead a counter will be able to track the total number of bytes being sent over this worker.

Reply:
> ... histograms make sense if there is a natural observation time to record the observation for. Since this is a continual stream, that doesn't exist here.

Didn't come across my mind but yeah we're working with websockets in this case - great point!

Reply:
Addressed in [7574f02](https://github.com/hashicorp/boundary/pull/1963/commits/7574f02e4e09da0681804e529875258b6db071cd)